src/snowflake/snowpark/session.py

```diff
@@ -437,7 +437,12 @@ def __init__(
 """
         self._session_stage = random_name_for_temp_object(TempObjectType.STAGE)
         self._stage_created = False
-        self._udf_registration = UDFRegistration(self)
+
+        if options is not None and options.get("local_testing", False):
+            self._udf_registration = MockUDFRegistration(self)
+        else:
+            self._udf_registration = UDFRegistration(self)
```

Comment:
this is for future discussion but want to collect your thoughts:

in general I want to avoid if-else checking (as few as possible) in the code to decide what implementation to use when switching between localtest and non-localtesting.

one potential approach is to subclass Session. do you see any other approaches here?

Reply:
I have some other ideas. Let's talk in person about it sometime.